src/api/index.ts

```diff
@@ -172,9 +172,21 @@ export function getAccountModule(
 export function view(
   request: Types.ViewRequest,
   nodeUrl: string,
+  ledgerVersion?: string,
 ): Promise<Types.MoveValue[]> {
   const client = new AptosClient(nodeUrl);
-  return withResponseError(client.view(request));
+  let parsedVersion = ledgerVersion;
+
+  // Handle non-numbers, to default to the latest ledger version
+  if (typeof ledgerVersion === "string" && isNaN(parseInt(ledgerVersion, 10))) {
+    parsedVersion = undefined;
+  }
+
+  if (ledgerVersion === null || ledgerVersion === "") {
+    return withResponseError(client.view(request, parsedVersion));
+  }
+
+  return withResponseError(client.view(request, ledgerVersion));
```

Comment:
What if the ledger version gets beyond the size of number, won't parseInt fail?

Reply:
The max safe integer for `number` is

9007199254740991

Now, let's say 1M transactions per second

9007199254740991 / 1000000 = 9007199254.740991
9007199254 / 60 seconds / 60 minutes / 24 hours / 365 days = 285.64 years

I think we'll be fine.